internal/interface/httpapi/api.go

```diff
@@ -150,7 +159,37 @@ func (a *Api) getPage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	http.Redirect(w, r, l, http.StatusSeeOther)
+	m := sync.Mutex{}
+	m.Lock()
+	a.RedirectRequestChannel <- RedirectRequestInput{w, r, l, &m}
+	m.Lock()
+}
+
+func CheckLink(in <-chan RedirectRequestInput) <-chan RedirectRequestInput {
+	out := make(chan RedirectRequestInput)
```

Comment:
This function has input and output of the same type. `out = RedirectRequestInput` is not very intuitive.